test/recipes/02-test_errstr.t

```diff
@@ -104,18 +104,27 @@ foreach my $errname (@posix_errors) {
         my @oerr = run(app([ qw(openssl errstr), sprintf("2%06x", $errnum) ]),
                        capture => 1);
         $oerr[0] =~ s|\R$||;
-        $oerr[0] =~ s|.*system library:||g; # The actual message is last
-
-        ok($oerr[0] eq $perr, "($errnum) '$oerr[0]' == '$perr'");
+        @oerr = split_error($oerr[0]);
+        ok($oerr[3] eq $perr, "($errnum) '$oerr[3]' == '$perr'");
     }
 }
 
 my @after = run(app([ qw(openssl errstr 2000080) ]), capture => 1);
 $after[0] =~ s|\R$||;
-$after[0] =~ s|.*system library:||g;
-ok($after[0] eq "reason(128)", "(128) '$after[0]' == 'reason(128)'");
+@after = split_error($after[0]);
+ok($after[3] eq "reason(128)", "(128) '$after[3]' == 'reason(128)'");
 
 my @zero = run(app([ qw(openssl errstr 2000000) ]), capture => 1);
 $zero[0] =~ s|\R$||;
-$zero[0] =~ s|.*system library:||g;
-ok($zero[0] eq "system library", "(0) '$zero[0]' == 'system library'");
+@zero = split_error($zero[0]);
+ok($zero[3] eq "system library", "(0) '$zero[3]' == 'system library'");
+
+# For an error string "error:xxxxxxxx:lib:func:reason", this returns
+# the following array:
+#
+# ( "xxxxxxxx", "lib", "func", "reason" )
+sub split_error {
+    my @erritems = $_[0] =~ m|error:([0-9A-F]+):([^:]*):([^:]*):(.*)|;
```

Comment:
why doesn't ```split /:/, $_[0]``` work?

Reply:
In case the reason contains a colon.

Reply:
```split /:/, $_[0], 4``` ?

Reply:
And anyway, mkerr can't handle reasons with colons in them.

Reply:
> `split /:/, $_[0], 4 ?`

...  I didn't know about the LIMIT argument.  Sure, that simplifies things!

> And anyway, mkerr can't handle reasons with colons in them.

This is about *system* errors.  You cannot predict what they will look like.

Reply:
Anyway, I've modified `split_error` and added some commentary

Reply:
I showed you something in Perl.  Amazing. :)